src/transmitter/index.ts

```diff
@@ -12,16 +12,40 @@ import {
   IDependencyGraphPayload,
 } from './types';
 import { getProxyAgent } from './proxy';
+import { queue } from 'async';
+
+interface HomebaseRequest {
+  method: NeedleHttpVerbs;
+  url: string;
+  payload:
+    IDependencyGraphPayload |
+    ScanResultsPayload |
+    IWorkloadMetadataPayload |
+    IDeleteWorkloadPayload;
+}
 
 const upstreamUrl = config.INTEGRATION_API || config.DEFAULT_KUBERNETES_UPSTREAM_URL;
 
+// Async queue wraps around the call to retryRequest in order to limit
+// the number of requests in flight to Homebase at any one time.
+const reqQueue = queue(async function(req: HomebaseRequest) {
+  return await retryRequest(req.method, req.url, req.payload);
```

Comment:
Might need to wrap in try-catch to avoid unhandled promise rejections

Reply:
The call to push is awaited and wrapped in try-catch. I'll need to check but that should avoid unhandled promise rejections:
https://github.com/snyk/kubernetes-monitor/pull/664/files#diff-d40843739012d28af7cb08fc87c96ded838baa28d5a6c617d41e75d9b5402d54R72

Reply:
Doesn't that line just enqueue the request? I think the callback here executes the request asynchronously with some sort of worker, but unfortunately `retryRequest()` can throw

Reply:
So, if I'm reading this page correctly: https://caolan.github.io/async/v3/global.html then an error should bubble up to the call to `push`.

I'll install it into a playground project and test it to verify though.

Reply:
I can confirm based on testing that this works as I had hoped.